server/schedulers/hot_region.go

```diff
@@ -882,37 +895,32 @@ func (bs *balanceSolver) betterThan(old *solution) bool {
 
 	if bs.cur.srcPeerStat != old.srcPeerStat {
 		// compare region
-
 		if toResourceType(bs.rwTy, bs.opTy) == writeLeader {
 			kind := statistics.GetRegionStatKind(statistics.Write, bs.firstPriority)
-			switch {
-			case bs.cur.srcPeerStat.GetLoad(kind) > old.srcPeerStat.GetLoad(kind):
-				return true
-			case bs.cur.srcPeerStat.GetLoad(kind) < old.srcPeerStat.GetLoad(kind):
-				return false
+			return bs.cur.srcPeerStat.GetLoad(kind) > old.srcPeerStat.GetLoad(kind)
+		}
+
+		firstCmp, secondCmp := bs.getRkCmpPriorities(old)
+		switch bs.cur.progressiveRank {
+		case -4: // isBetter(firstPriority) && isBetter(secondPriority)
+			if firstCmp != 0 {
+				return firstCmp > 0
 			}
-		} else {
-			firstCmp, secondCmp := bs.getRkCmpPriorities(old)
-			switch bs.cur.progressiveRank {
-			case -2: // greatDecRatio < firstPriorityDecRatio <= minorDecRatio && secondPriorityDecRatio <= greatDecRatio
-				if secondCmp != 0 {
-					return secondCmp > 0
-				}
-				if firstCmp != 0 {
-					// prefer smaller first priority rate, to reduce oscillation
-					return firstCmp < 0
-				}
-			case -3: // firstPriorityDecRatio <= greatDecRatio && secondPriorityDecRatio <= greatDecRatio
-				if secondCmp != 0 {
-					return secondCmp > 0
-				}
-				fallthrough
-			case -1: // firstPriorityDecRatio <= greatDecRatio
-				if firstCmp != 0 {
-					// prefer region with larger first priority rate, to converge faster
-					return firstCmp > 0
-				}
+			return secondCmp > 0
+		case -3, -1:
```

Comment:
I think we need to split these two cases. -3 should be more similar with -2

Reply:
I wrote them separately at first, but the two seem to have the same processing logic at the moment.

Reply:
I think it will be friendly to read code and modify if split these cases.